### ISSUE TITLE
Linter errors link to wrong file location when in an umbrella application

### ISSUE THREAD

**Issue by brainlid** [AtomLinter/linter-elixirc#21]
While working through the Elixir [Getting Started guide (for mixes)](http://elixir-lang.org/getting-started/mix-otp/introduction-to-mix.html), when I reached the step for [Dependencies and Umbrella Apps](http://elixir-lang.org/getting-started/mix-otp/dependencies-and-umbrella-apps.html#umbrella-projects), I ended up with an umbrella app and two child mix projects.

This caused issues for the linter. I shutdown atom, and did a `mix compile` for the umbrella project. Reopened Atom (on the umbrella project) and intentionally create an error in a child project.

The arrows highlight the errant code, the true location of the file in the project and the displayed linter error location. The errant code is not highlighted as being wrong.

![selection_010](https://cloud.githubusercontent.com/assets/808984/11432367/3398a620-946a-11e5-8c7a-de678b3d87df.png)

Double clicking the "1 issue" opens the invalid file as a new tab.

![selection_011](https://cloud.githubusercontent.com/assets/808984/11432368/339a1636-946a-11e5-995a-773d903ae59e.png)

**Comment by Arcanemagus** [comment]
Can you see if you are still running into this issue in v1.1.6?

**Comment by brainlid** [comment]
Yes, I just confirmed this is still a bug. 

Atom: 1.7.4
linter-elixirc: 1.1.8

**Comment by Arcanemagus** [comment]
Can you open Atom's Developer Tools and place a breakpoint [here](https://github.com/AtomLinter/linter-elixirc/blob/v1.1.8/lib/init.coffee#L130), and then paste the contents of `resultstring` back here?

**Comment by brainlid** [comment]
Here's the results:

```
resultString
"== Compilation error on file lib/kv.ex ==
** (CompileError) lib/kv.ex:6: undefined function start_link/0
    (stdlib) lists.erl:1337: :lists.foreach/2
    (stdlib) erl_eval.erl:670: :erl_eval.do_apply/6
"
```

The correct full file path was output to the console from this file (autocomplete-elixir plugin).

`file:///home/mark/.atom/packages/autocomplete-elixir/lib/alchemide/wrapper.coffee`

The logged console message:

```
File load /home/mark/dev/elixir/playing/kv_umbrella/apps/kv/lib/kv.ex -> Error {:error, %CompileError{description: "undefined function start_link/0", file: "dev/elixir/playing/kv_umbrella/apps/kv/lib/kv.ex", line: 6}}<><>
```

So hopefully that helps as a reference of how to get a more complete file path to the error.

**Comment by Arcanemagus** [comment]
Hmmmm, And you had the `kv_umbrella` folder open as the Atom project, like in your first screenshot above?

`elixirc` only gives relative paths to the file based on where it was ran. We are running it from the project root, as Atom defines it (the folder you have open in Atom essentially). In this case the above text _should_ only have been output if you had the `apps` folder open... but then it would have corrected the path to the full path correctly if that was the case :confused:.

Can you put a breakpoint [here](https://github.com/AtomLinter/linter-elixirc/blob/v1.1.8/lib/init.coffee#L45) and see what it is getting for the `projectPath`?

**Comment by brainlid** [comment]
`projPath`
"/home/mark/dev/elixir/playing/kv_umbrella"

`editorPath`
"/home/mark/dev/elixir/playing/kv_umbrella/apps/kv/lib/kv.ex"

I've attached the whole project. [kv_umbrella.zip](https://github.com/AtomLinter/linter-elixirc/files/287471/kv_umbrella.zip)

Nothing sensitive. Just created while following along with [this guide](http://elixir-lang.org/getting-started/mix-otp/dependencies-and-umbrella-apps.html#umbrella-projects). 

My OS is Arch Linux. Elixir is 1.2.5.

**Comment by 2c2c** [comment]
bumped into the same thing on windows10 following the same guide

**Comment by sjmueller** [comment]
same thing happening to me

**Comment by Arcanemagus** [comment]
Looks like this was fixed in #82.
